algorithms.py

```diff
@@ -1148,6 +1159,20 @@ def print_heaps(self, L):
                 u = u.next
             print(s)
 
+    def census_event(self, time):
+        for pop in self.P:
+            # pop.print_state()
+            for ancestor in pop.iter_ancestors():
+                seg = ancestor
+                while seg is not None:
+                    self.flush_edges()
+                    u = self.tables.nodes.add_row(
```

Comment:
Ah, also, adding too many nodes. This is one node per segment, definitely too much!

Reply:
> This is one node per segment, definitely too much!

Perhaps I should go back and read the msprime paper in a bit more detail because I still don't quite understand this. Here's some output of this current version of `algorithms.py` with one census event, and census nodes 9, 10, 11 and 12. It looks okay to me.

are you saying that the census nodes on the last tree (10, 12) should be the same as on the first few trees (9, 11)?

![Screen Shot 2019-08-23 at 10 56 50 am](https://user-images.githubusercontent.com/17755074/63584468-ec315b00-c594-11e9-83ef-e911a1521dba.png)